src/main/java/org/stellar/sdk/Server.java

```diff
@@ -115,41 +109,22 @@ public TransactionsRequestBuilder transactions() {
      * @throws IOException
      */
     public SubmitTransactionResponse submitTransaction(Transaction transaction) throws IOException {
-        URI transactionsURI;
-        try {
-            transactionsURI = new URIBuilder(serverURI).setPath("/transactions").build();
-        } catch (URISyntaxException e) {
-            throw new AssertionError(e);
-        }
-        HttpPost submitTransactionRequest = new HttpPost(transactionsURI);
-
-        List<NameValuePair> params = new ArrayList<NameValuePair>();
-        params.add(new BasicNameValuePair("tx", transaction.toEnvelopeXdrBase64()));
-        submitTransactionRequest.setEntity(new UrlEncodedFormEntity(params));
-
-        HttpResponse response = httpClient.execute(submitTransactionRequest);
-        HttpEntity entity = response.getEntity();
-
-        if (entity != null) {
-            InputStream responseStream = entity.getContent();
-            try {
-                StringWriter writer = new StringWriter();
-                IOUtils.copy(responseStream, writer, StandardCharsets.UTF_8);
-                String responseString = writer.toString();
-                SubmitTransactionResponse submitTransactionResponse = GsonSingleton.getInstance().fromJson(responseString, SubmitTransactionResponse.class);
-                return submitTransactionResponse;
-            } finally {
-                responseStream.close();
-            }
+        HttpUrl transactionsURI = serverURI.newBuilder().addPathSegment("transactions").build();
+
+        RequestBody requestBody = new MultipartBody.Builder()
```

Comment:
RequestBody formBody = new FormBody.Builder().add("tx", transaction.toEnvelopeXdrBase64()).build();
Should suffice

Reply:
Better, thanks!